server/src/main/java/org/elasticsearch/index/search/QueryStringQueryParser.java

```diff
@@ -525,7 +525,8 @@ private Query getPrefixQuerySingle(String field, String termStr) throws ParseExc
 
     private Query getPossiblyAnalyzedPrefixQuery(String field, String termStr) throws ParseException {
         if (analyzeWildcard == false) {
-            return super.getPrefixQuery(field, termStr);
+            return currentFieldType.prefixQuery(getAnalyzer().normalize(field, termStr).utf8ToString(),
```

Comment:
`getPossiblyAnalyzedPrefixQuery` is only called from `getPrefixQuerySingle`, which already checks to see if this is a `StringFieldType` and short-circuits if so, see: https://github.com/elastic/elasticsearch/blob/262b9d34ca31fb7a25531991a51992cb5b7c13bd/server/src/main/java/org/elasticsearch/index/search/QueryStringQueryParser.java#L510

So I think double-normalization is already accounted for?

Reply:
I'm hesitant to adding a boolean which would make things more complex. I'd expect normalization to be idempotent so that shouldn't matter?

Reply:
> So I think double-normalization is already accounted for?

`KeywordFieldType` extends `StringFieldType` so the double-normalization occurs. 

> I'd expect normalization to be idempotent so that shouldn't matter?

It can be difficult to detect and some "official" char filters (PatternReplace?) could create weird double-normalization ? Another solution could be to change the `currentFieldType instanceof StringFieldType == false` to `currentFieldType.tokenized() == false` ?

Reply:
> Another solution could be to change the currentFieldType instanceof StringFieldType == false to currentFieldType.tokenized() == false 

++, I've changed it to this